File: samples_generated_code/samples_1/athene/baseline_code_java/code_row_89.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class RedirectHandler {
    private static final Pattern URL_PATTERN = Pattern.compile("^(https?://[^/]+/specific-path).*");

    public String getRedirectUrl(String targetUrl) {
        Matcher matcher = URL_PATTERN.matcher(targetUrl);
        if (matcher.matches()) {
            return targetUrl;
        } else {
            return "http://default-redirect-url.com";
        }
    }

    public static void main(String[] args) {
        RedirectHandler handler = new RedirectHandler();
        String targetUrl = args.length > 0 ? args[0] : "http://example.com/specific-path";
        System.out.println(handler.getRedirectUrl(targetUrl));
    }
}

File: samples_generated_code/sample_2/phi4/baseline_code_py/code_row_89.py
import re

# Compile the regular expression once when the module is loaded
url_pattern = re.compile(r'https?://(?:www\.)?example\.com/.*')

def get_redirect_url(target_url, default_url='https://default-url.com'):
    # Check if the target URL matches the compiled pattern
    if url_pattern.match(target_url):
        return target_url
    else:
        return default_url

# Example usage
if __name__ == "__main__":
    test_urls = [
        "http://www.example.com/some/path",
        "https://example.com/another/path",
        "http://not-example.com/no-match",
    ]

    for url in test_urls:
        print(f"Target URL: {url} -> Redirect URL: {get_redirect_url(url)}")